test/functional/test_framework/util.py

```diff
@@ -50,6 +50,11 @@ def assert_equal(thing1, thing2, *args):
         raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
 
 
+def assert_not_equal(thing1, thing2, *args):
+    if thing1 == thing2 or any(thing1 != arg for arg in args):
```

Comment:
Should it possibly be:

```suggestion
    if thing1 == thing2 or any(thing1 == arg for arg in args):
```

?

Reply:
Good catch @kiminuo thanks you!